diff -up gnome-terminal-2.31.1/po/POTFILES.in.potfiles gnome-terminal-2.31.1/po/POTFILES.in
--- gnome-terminal-2.31.1/po/POTFILES.in.potfiles	2010-05-04 00:41:15.490932905 -0400
+++ gnome-terminal-2.31.1/po/POTFILES.in	2010-05-04 00:41:26.265182941 -0400
@@ -17,8 +17,6 @@ src/terminal-accels.c
 src/terminal-app.c
 src/terminal.c
 src/terminal-encoding.c
-src/terminal-find.c
-src/terminal-find.glade
 src/terminal-options.c
 src/terminal-profile.c
 src/terminal-screen.c
