diff -up gnome-terminal-2.33.3/configure.ac.builder gnome-terminal-2.33.3/configure.ac
--- gnome-terminal-2.33.3/configure.ac.builder	2011-01-07 17:37:33.449993002 -0500
+++ gnome-terminal-2.33.3/configure.ac	2011-01-07 17:37:49.942992152 -0500
@@ -130,10 +130,7 @@ AC_SUBST([GLIB_GENMARSHAL])
 GLIB_MKENUMS="$($PKG_CONFIG --variable=glib_mkenums glib-2.0)"
 AC_SUBST([GLIB_MKENUMS])
 
-case "$with_gtk" in
-  2.0) AC_PATH_PROG([GTK_BUILDER_CONVERT],[gtk-builder-convert],[false]) ;;
-  3.0) AC_PATH_PROG([GTK_BUILDER_CONVERT],[gtk-builder-convert-3.0],[false]) ;;
-esac
+AC_PATH_PROG([GTK_BUILDER_CONVERT],[gtk-builder-convert],[false])
 
 if test "$GTK_BUILDER_CONVERT" = "false"; then
   AC_MSG_ERROR([gtk-builder-convert not found])
