diff -up gnome-terminal-2.33.3/src/Makefile.am.libs gnome-terminal-2.33.3/src/Makefile.am
--- gnome-terminal-2.33.3/src/Makefile.am.libs	2011-01-07 22:19:10.168992663 -0500
+++ gnome-terminal-2.33.3/src/Makefile.am	2011-01-07 22:19:47.375993276 -0500
@@ -75,7 +75,8 @@ gnome_terminal_CFLAGS = \
 gnome_terminal_LDFLAGS =
 
 gnome_terminal_LDADD = \
-	$(TERM_LIBS)
+	$(TERM_LIBS) \
+	-lX11
 
 if WITH_SMCLIENT
 gnome_terminal_SOURCES += \
