diff -up gnome-terminal-2.33.0/configure.ac.gtk3-build gnome-terminal-2.33.0/configure.ac
--- gnome-terminal-2.33.0/configure.ac.gtk3-build	2010-10-07 13:02:12.307779002 -0400
+++ gnome-terminal-2.33.0/configure.ac	2010-10-07 13:05:13.335779007 -0400
@@ -129,7 +129,7 @@ AC_SUBST([GLIB_GENMARSHAL])
 GLIB_MKENUMS="$($PKG_CONFIG --variable=glib_mkenums glib-2.0)"
 AC_SUBST([GLIB_MKENUMS])
 
-AC_PATH_PROG([GTK_BUILDER_CONVERT],[gtk-builder-convert],[false])
+AC_PATH_PROGS([GTK_BUILDER_CONVERT],[gtk-builder-convert] [gtk-builder-convert-3.0],[false])
 if test "$GTK_BUILDER_CONVERT" = "false"; then
   AC_MSG_ERROR([gtk-builder-convert not found])
 fi
