diff -up gnome-terminal-2.24.1/src/gnome-terminal.schemas.in.tab-switching gnome-terminal-2.24.1/src/gnome-terminal.schemas.in
--- gnome-terminal-2.24.1/src/gnome-terminal.schemas.in.tab-switching	2008-10-21 22:14:28.000000000 -0400
+++ gnome-terminal-2.24.1/src/gnome-terminal.schemas.in	2008-10-21 22:14:49.000000000 -0400
@@ -921,6 +921,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
       <default>&lt;Alt&gt;1</default>
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 1</short>
+         <long>
+          Keyboard shortcut key for switch to tab 1. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
@@ -929,6 +938,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
       <default>&lt;Alt&gt;2</default>
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 2</short>
+         <long>
+          Keyboard shortcut key for switch to tab 2. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
@@ -937,6 +955,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
       <default>&lt;Alt&gt;3</default>
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 3</short>
+         <long>
+          Keyboard shortcut key for switch to tab 3. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
@@ -945,6 +972,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
       <default>&lt;Alt&gt;4</default>
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 4</short>
+         <long>
+          Keyboard shortcut key for switch to tab 4. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
@@ -953,6 +989,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
       <default>&lt;Alt&gt;5</default>
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 5</short>
+         <long>
+          Keyboard shortcut key for switch to tab 5. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
@@ -961,6 +1006,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
       <default>&lt;Alt&gt;6</default>
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 6</short>
+         <long>
+          Keyboard shortcut key for switch to tab 6. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
@@ -969,6 +1023,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
       <default>&lt;Alt&gt;7</default>
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 7</short>
+         <long>
+          Keyboard shortcut key for switch to tab 7. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
@@ -977,6 +1040,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
       <default>&lt;Alt&gt;8</default>
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 8</short>
+         <long>
+          Keyboard shortcut key for switch to tab 8. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
@@ -985,6 +1057,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
       <default>&lt;Alt&gt;9</default>
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 9</short>
+         <long>
+          Keyboard shortcut key for switch to tab 9. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
@@ -993,6 +1074,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
 <!-- no default -->
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 10</short>
+         <long>
+          Keyboard shortcut key for switch to tab 10. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
@@ -1001,6 +1091,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
 <!-- no default -->
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 11</short>
+         <long>
+          Keyboard shortcut key for switch to tab 11. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
@@ -1009,6 +1108,15 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
 <!-- no default -->
+      <locale name="C">
+         <short>Keyboard shortcut to switch to tab 12</short>
+         <long>
+          Keyboard shortcut key for switch to tab 12. Expressed as a string
+          in the same format used for GTK+ resource files.
+          If you set the option to the special string "disabled", then there
+          will be no keyboard shortcut for this action.
+         </long>
+      </locale>
     </schema>
 
     <schema>
